src/seismometer/core/nbhost.py

```diff
@@ -0,0 +1,22 @@
+from enum import Enum
+
+import psutil
+
+
+class NotebookHost(Enum):
+    JUPYTER_LAB = 1
+    JUPYTER_HUB = 2
+    VSCODE = 3
+    OTHER = -1
+
+    @classmethod
+    def get_current_host(cls):
+        cmds = [x for x in psutil.Process().parent().cmdline()]
+
+        if any("jupyter-lab" in cmd for cmd in cmds):
+            return cls.JUPYTER_LAB
+        elif any("jupyterhub" in cmd for cmd in cmds):
+            return cls.JUPYTER_HUB
+        elif any("vscode-server" in cmd for cmd in cmds):
```

Comment:
What is actually being matched? And can we more explicit about it (exact match list of pieces)?  

From my test cases, the types of maching seems varied.
 
jupyterhub (jupyterhub-singleuser) is a portion of the default startup command; default seems reused enough, but the partial makes me worry about things like a container or path that happens to have it - made worse by the other comment if we can only be sure about "jupyter".  
vscode-server is just a directory, the worry on this one is more that parents of jupyter-lab might include this.

Reply:
updated to env vars to fix this.